### PR TITLE
add `:wait` parameter for Docker::Exec#start!

### PR DESCRIPTION
**Pull request by vexus2**
I just add `:wait` parameter for Docker::Exec#start!

### CODE REVIEW

**Comment by tlunter** [comment]
Please remove the increment version.  We handle that on master.

Looks like your VCR wasn't added.  Can you add that as well?

File: spec/docker/exec_spec.rb [hunk 1]
```diff
@@ -138,6 +138,17 @@
...
 
+    context 'when :wait set with some values' do
+      subject {
+        described_class.create('Container' => container.id, 'Cmd' => %w[date])
+      }
+      after { container.kill!.remove }
+
+      it 'returns empty stdout/stderr messages with exitcode', :vcr do
+        expect(subject.start!(:wait => 100)).to eq([[],[], 0])
+      end
```
Comment (tlunter) [inline spec/docker/exec_spec.rb]: Can you also add a test case for when the command is something like `["/bin/bash","-lc","sleep 2"]` with a timeout of 1? If VCR doesn't seem to want to record that, don't worry about adding it.

**Comment by vexus2** [comment]
@tlunter thank you for response!
i add vcr record and fix that.

File: spec/docker/exec_spec.rb [hunk 2]
```diff
@@ -138,6 +138,33 @@
...
+        described_class.create(
+            'Container'    => container.id,
+            'AttachStdout' => true,
+            'Cmd'          => ['bash', '-c', 'sleep 2; echo hello']
+        )
+      }
+      after { container.kill!.remove }
+
+      it 'raises an error', :vcr do
+        pending "VCR doesn't like to record errors"
```
Comment (vexus2) [inline spec/docker/exec_spec.rb]: VCR can't record TimeoutError so I add pending here like that.
https://github.com/swipely/docker-api/blob/bd1fc83d60d37f9fc1d9fa7e0ead4216f78e9b8d/spec/docker/container_spec.rb#L530-L534

**Comment by vexus2** [comment]
@tlunter How about here?!

**Comment by tlunter** [comment]
@vexus2 sorry about that, been busy with work.  Looks great. Thanks.

**Comment by tlunter** [comment]
Just pushed v1.23.0 with this!

**Comment by hyz0906** [comment]
The `:wait` parameter is not passed to the exec.start!().So it doesn't take affect when call `container.exec` with `:wait`
```
    start_opts = {
      :tty => tty,
      :stdin => stdin,
      :detach => detach
    }

    if detach
      instance.start!(start_opts)
      return instance
    else
      instance.start!(start_opts, &block)
    end
```
[https://github.com/swipely/docker-api/blob/master/lib/docker/container.rb#L77](url)

**Pull request merged**